libyul/optimiser/StackCompressor.cpp

```diff
@@ -249,13 +249,15 @@ std::tuple<bool, Block> StackCompressor::run(
 	);
 	bool usesOptimizedCodeGenerator = false;
 	bool simulateFunctionsWithJumps = true;
+	size_t reachableStackDepth = 16;
 	if (auto evmDialect = dynamic_cast<EVMDialect const*>(_object.dialect()))
 	{
 		usesOptimizedCodeGenerator =
 			_optimizeStackAllocation &&
 			evmDialect->evmVersion().canOverchargeGasForCall() &&
 			evmDialect->providesObjectAccess();
 		simulateFunctionsWithJumps = !evmDialect->eofVersion().has_value();
+		reachableStackDepth = evmDialect->reachableStackDepth();
 	}
```

Comment:
This seems a bit contradictory. On one hand `reachableStackDepth()` is only defined in the EVM dialect, which makes sense, because some targets may not be stack machines. But on the other hand we still run `StackLayoutGenerator` for those other dialects, which implicitly assumes that they do provide a stack.

So which one is it? IMO we should either put `reachableStackDepth()` in `Dialect` or stop pretending we can support anything other than the EVM dialect in `StackCompressor` :)

Reply:
It's a bit weird code-flow wise - but we can only ever run the stack layout generator for the EVM dialect here anyways (note that ``usesOptimizedCodeGenerator`` is always ``false`` otherwise and we only run the old ``CompilabilityChecker`` then - which does nothing for non-evm dialects, but could, in principle, be extended to other dialects, although I don't see that happening).

I'd postpone refactoring this further until we drop ancient EVM versions, which would generally simplify this quite a bit (and then ultimately make it evm-dialect specific)

Reply:
If we stop running it on EOF code anyways, the problem will vanish entirely :-).